include/kodlab_mjbots_sdk/lcm_subscriber.h

```diff
@@ -1,85 +1,121 @@
+   * @tparam Message LCM message type for channel
+   * @param channel_name channel name
+   * @param handler `LcmMessageHandler`-inherited message handler object
+   * @return generated `lcm::Subscription` object
    */
-  void HandleMsg(const lcm::ReceiveBuffer *rbuf,
-                 const std::string &chan,
-                 const MessageClass *msg);
+  template<class Message>
+  lcm::Subscription *AddSubscription(std::string channel_name,
+                                     LcmMessageHandler<Message> &handler) {
+    auto sub = lcm_.subscribe(channel_name,
+                              &LcmMessageHandler<Message>::HandleMessage,
+                              &handler);
+    subs_.emplace(channel_name, sub);
+    return sub;
+  }
 
-  /*!
-   * @brief subscribes to the lcm channel using handle message and handles ctrl c detection
+  /**
+   * @brief Remove a subscription by channel name
+   * @param channel_name channel name of subscription to be removed
    */
-  void Run() override;
+  void RemoveSubscription(const std::string& channel_name);
 
-  std::string channel_name_;
+  /**
+   * @brief Accessor for `lcm::Subscription` objects
+   * @param channel_name channel name
+   * @return `lcm::Subscription` object on corresponding channel
+   */
+  [[nodiscard]] const lcm::Subscription *get_subscription(const std::string &channel_name) const;
+
+ private:
+
+  /**
+   * @brief LCM object
+   */
   lcm::LCM lcm_;
+
+  /**
+   * @brief Map of channel names to corresponding `lcm::Subscription` objects
+   */
+  std::map<std::string, lcm::Subscription *> subs_;
+
+  /**
+   * @brief Waits for LCM messages and exits when ctrl+c detected
+   */
+  void Run() override;
+
 };
 
-/************************************Implementation********************************************************************/
-template<class msg_type>
-void LcmSubscriber<msg_type>::Run() {
-  lcm_.subscribe(channel_name_, &LcmSubscriber::HandleMsg, this);
-  mutex_.unlock(); // Ensures mutex is unlocked
-  std::cout << "Subscribing to " << channel_name_ << std::endl;
-  while (!CTRL_C_DETECTED) {
-    lcm_.handleTimeout(1000);
-  }
-}
 
-template<class msg_type>
-LcmSubscriber<msg_type>::LcmSubscriber(int realtime_priority, int cpu, std::string channel_name):
-    channel_name_(channel_name) {
+////////////////////////////////////////////////////////////////////////////////
+// Implementation                                                             //
+////////////////////////////////////////////////////////////////////////////////
+
+LcmSubscriber::LcmSubscriber(int realtime_priority, int cpu) {
   cpu_ = cpu;
   realtime_priority_ = realtime_priority;
-  if (!channel_name_.empty())
-    Start();
+  Start();  // FIXME: Call outside constructor
+}
+
+void LcmSubscriber::RemoveSubscription(const std::string& channel_name) {
+  lcm_.unsubscribe(subs_[channel_name]);
+  subs_.erase(subs_.find(channel_name));
```

Comment:
This works but I think that erase also takes just the key itself ie
```suggestion
  subs_.erase(channel_name);
```
which is easier to read

Reply:
Reading more about it it seems slightly more complicated, but since we are using the [] operator immediately above we have inherently assumed that channel name will be in subs_ map. 

Likely though we should check it is in the map and then interact with it? https://stackoverflow.com/a/33239463

Reply:
Handled invalid channel entries in 5accfd78db2fcd592f411818e7780800fed596e8.